src/ftest.jl

```diff
@@ -28,6 +35,34 @@ _diffn(t::NTuple{N, T}) where {N, T} = ntuple(i->t[i]-t[i+1], N-1)
 
 _diff(t::NTuple{N, T}) where {N, T} = ntuple(i->t[i+1]-t[i], N-1)
 
+"""
+    ftest(mod::LinearModel)
+
+Prints the F-test against the null model (`Result ~ 1`).
+
+```jldoctest
+julia> dat = DataFrame(Result=[1.1, 1.2, 1, 2.2, 1.9, 2, 0.9, 1, 1, 2.2, 2, 2],
+                       Treatment=[1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2]);
+
+julia> model = lm(@formula(Result ~ 1 + Treatment), dat);
+
+julia> ftest(model.model)
+F-test against the null model:
+F-statistic: 241.62 on 12 and 1 degrees of freedom, p-value: 0.0502
+```
+"""
+function ftest(mod::LinearModel)
+    rss = deviance(mod)
+    tss = nulldeviance(mod)
+
+    n = nobs(mod)
+    p = dof(mod) - 2 # -2 for dispersion parameter and intercept
+    fstat = ((tss - rss) * (n - p - 1)) / (rss * p)
+    fdist = FDist(n, p)
```

Comment:
This isn't the correct formula, is it? I'd expect something like `FDist(p-1, p)`.
Also this doesn't give the same result as the existing `ftest` method for the example given above in the docstring. Have you checked this against other implementations? BTW, can you add tests for the new method?